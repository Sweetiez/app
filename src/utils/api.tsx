import Config from 'react-native-config';
import {Platform} from 'react-native';

export const getUrl = (path: string) => {
  if (Platform.OS === 'ios') {
    return Config.API_IOS_LOCAL_URL + path;
  } else {
    return Config.API_URL + path;
  }
};

export const buildHeaders = token => {
  return {
    Accept: 'application/json',
    'Content-Type': 'application/json',
    Authorization: token ? 'Bearer ' + token : undefined,
  };
};

export const buildRequest = async (
  method,
  url,
  data,
  onError,
  onSuccess,
  token,
) => {
  const headers = buildHeaders(token);
  try {
    const response = await fetch(getUrl(url), {
      method: method,
      headers,
      body: JSON.stringify(data),
    });
    if (response.status === 200 || response.status === 201) {
      return onSuccess(response);
    } else {
      return onError();
    }
  } catch (error) {
    return onError();
  }
};
