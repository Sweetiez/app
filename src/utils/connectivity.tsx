import NetInfo from '@react-native-community/netinfo';

export const checkConnectivity = async () => {
  const status = await NetInfo.fetch();
  return status.isConnected;
};
