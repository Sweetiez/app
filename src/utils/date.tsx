//@ts-ignore
import moment from 'moment';

export const formatDate = (date: string) => {
  return moment(date).format('DD/MM/YYYY');
};
export const formatDashDate = (date: string) => {
  return moment(date).format('YYYY-MM-DD');
};
