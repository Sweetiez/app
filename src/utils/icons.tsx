// @ts-nocheck
import React from 'react';
import Pen from './../assets/icons/pen.svg';
import Minus from './../assets/icons/minus.svg';
import Plus from './../assets/icons/plus.svg';
import Add from './../assets/icons/add.svg';
import Back from './../assets/icons/arrow-back.svg';
import CookBook from './../assets/icons/cook-book.svg';
import Cake from './../assets/icons/cake.svg';
import Event from './../assets/icons/event.svg';
import Account from './../assets/icons/account.svg';
import Eye from './../assets/icons/eye.svg';
import Interrogation from './../assets/icons/interrogation.svg';
import Cart from './../assets/icons/cart.svg';
import Trash from './../assets/icons/trash.svg';
import NoImage from './../assets/icons/no-image.svg';
import Edit from './../assets/icons/edit.svg';
import Warning from './../assets/icons/warning.svg';
import Orders from './../assets/icons/orders.svg';
import Filter from './../assets/icons/filter.svg';
import Cross from './../assets/icons/cross.svg';
import Circle from './../assets/icons/circle.svg';
import colors from '../assets/colors';

const getIcons = (iconName, color = colors.white, size = 18) => {
  const props = {
    width: size,
    height: size,
    color: color,
  };
  switch (iconName) {
    case 'pen':
      return <Pen {...props} />;
    case 'eye':
      return <Eye {...props} />;
    case 'minus':
      return <Minus {...props} />;
    case 'plus':
      return <Plus {...props} />;
    case 'add':
      return <Add {...props} />;
    case 'back':
      return <Back {...props} />;
    case 'cake':
      return <Cake {...props} />;
    case 'event':
      return <Event {...props} />;
    case 'cookBook':
      return <CookBook {...props} />;
    case 'account':
      return <Account {...props} />;
    case 'interrogation':
      return <Interrogation {...props} />;
    case 'cart':
      return <Cart {...props} />;
    case 'trash':
      return <Trash {...props} />;
    case 'noImage':
      return <NoImage {...props} />;
    case 'edit':
      return <Edit {...props} />;
    case 'warning':
      return <Warning {...props} />;
    case 'orders':
      return <Orders {...props} />;
    case 'filter':
      return <Filter {...props} />;
    case 'cross':
      return <Cross {...props} />;
    case 'circle':
      return <Circle {...props} />;
  }
};

export default getIcons;
