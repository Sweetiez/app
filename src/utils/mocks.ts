jest.mock('./../assets/icons/pen.svg', () => 'Pen');
jest.mock('./../assets/icons/star.svg', () => 'Star');
jest.mock('./../assets/icons/plus.svg', () => 'Plus');
jest.mock('./../assets/icons/minus.svg', () => 'Minus');
jest.mock('./../assets/icons/add.svg', () => 'Add');
jest.mock('./../assets/icons/arrow-back.svg', () => 'Back');
jest.mock('./../assets/icons/cook-book.svg', () => 'CookBook');
jest.mock('./../assets/icons/cake.svg', () => 'Cake');
jest.mock('./../assets/icons/event.svg', () => 'Event');
jest.mock('./../assets/icons/eye.svg', () => 'Eye');
jest.mock('./../assets/icons/trash.svg', () => 'Trash');
jest.mock('./../assets/icons/no-image.svg', () => 'NoImage');
jest.mock('./../assets/icons/edit.svg', () => 'Edit');
jest.mock('./../assets/icons/warning.svg', () => 'Warning');

jest.mock('react-i18next', () => ({
    useTranslation: () => {
        return {
            t: (str) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
}))



jest.mock('react-native-picker-select', () => 'RNPickerSelect')
jest.mock('react-native-config', () => 'Config')