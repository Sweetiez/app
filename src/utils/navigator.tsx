import getIcons from './icons';

const getTabIcon = (route, focused, color, size, t) => {
  switch (route.name) {
    case t('tabBar.shop'):
      return getIcons('cake', color, 24);
    case t('tabBar.recipes'):
      return getIcons('cookBook', color, 24);
    case t('tabBar.events'):
      return getIcons('event', color, 24);
    case t('tabBar.account'):
      return getIcons('account', color, 24);
    case t('tabBar.cart'):
      return getIcons('cart', color, 24);
  }
};

export default getTabIcon;
