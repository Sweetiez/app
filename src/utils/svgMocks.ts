
jest.mock('./../assets/icons/pen.svg', () => 'Pen');
jest.mock('./../assets/icons/star.svg', () => 'Star');
jest.mock('./../assets/icons/plus.svg', () => 'Plus');
jest.mock('./../assets/icons/minus.svg', () => 'Minus');
jest.mock('./../assets/icons/add.svg', () => 'Add');