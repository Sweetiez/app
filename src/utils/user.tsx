export function cleanToken(token: string): string {
  return token?.slice(7);
}
