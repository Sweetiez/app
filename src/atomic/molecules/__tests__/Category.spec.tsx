import * as React from 'react';
import renderer from 'react-test-renderer';
import '../../../utils/mocks';
import Category from '../Category';

describe('Category', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Category
          isChecked={false}
          setChecked={jest.fn()}
          label={'this is a checkbox'}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders checked', () => {
    const tree = renderer
      .create(
        <Category
          isChecked={true}
          setChecked={jest.fn()}
          label={'this is a checkbox'}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
