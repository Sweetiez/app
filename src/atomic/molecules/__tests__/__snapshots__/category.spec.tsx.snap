// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category renders checked 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginTop": 10,
        "width": "100%",
      },
    ]
  }
>
  <View
    accessibilityRole="checkbox"
    accessibilityState={
      Object {
        "checked": false,
        "disabled": false,
      }
    }
    accessibilityValue={
      Object {
        "text": "off",
      }
    }
    accessible={true}
    pointerEvents="auto"
  >
    <RNCCheckbox
      forwardedRef={null}
      onValueChange={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "height": 32,
            "width": 32,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "width": 10,
        },
      ]
    }
  />
  <Text>
    this is a checkbox
  </Text>
</View>
`;

exports[`Category renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginTop": 10,
        "width": "100%",
      },
    ]
  }
>
  <View
    accessibilityRole="checkbox"
    accessibilityState={
      Object {
        "checked": false,
        "disabled": false,
      }
    }
    accessibilityValue={
      Object {
        "text": "off",
      }
    }
    accessible={true}
    pointerEvents="auto"
  >
    <RNCCheckbox
      forwardedRef={null}
      onValueChange={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "height": 32,
            "width": 32,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "width": 10,
        },
      ]
    }
  />
  <Text>
    this is a checkbox
  </Text>
</View>
`;
